Add AUTHORS & Script auto AUTHORS generate_authors

BDadmehr0 opened the pull request:
@jbampton 

AUTHORS file look like [mruby/AUTHORS](https://github.com/mruby/mruby/blob/master/AUTHORS)

BDadmehr0 committed:
Add files via upload

jbampton reviewed (commented):
You should remove all the unneeded code comments

jbampton reviewed (commented):
Seems that some entries are duplicated

Perhaps a groupby / merge is missing for commit author ??

jbampton reviewed (commented):
If you run pre-commit locally it will help format your files to be compliant.

Otherwise you can run "black" on your Python files.

And make sure there is no trailing whitespace in the Authors file.

You can see these issues in the pre-commit failure workflow.

https://black.readthedocs.io/en/stable/

BDadmehr0 committed:
Update generate_authors.py

run black and remove comments
list with user's usernames

BDadmehr0 committed:
Update AUTHORS

BDadmehr0 commented:
fixed @jbampton

jbampton reviewed (commented):
We have to decide what to do with the new bandit issues shown in the pre-commit failure workflow

https://github.com/PyCQA/bandit

https://bandit.readthedocs.io/en/latest/

jbampton commented:
It seems the commit totals are not 100% correct.

@AmirTallap would you like to review this PR ?

Is there an easier way to do this ?

BDadmehr0 committed:
Update generate_authors.py

- Using `shutil.which("git")`: Resolves **B607** (partial executable path)
- Checking for the existence of `git` and handling errors: Increases security and prevents crashes
- Error handling in `subprocess.check_output`: Addresses **B603** and follows better security practices
- Keeping `subprocess` usage only with safe paths and hardcoded arguments: Fully resolves **B603**, **B607**, and **B404**

jbampton reviewed (commented):
We should not store the Python script in the repo root.  We try to minimize the amount of files in the root.

There is a `tools` folder perhaps we should make a sub directory there called `scripts` so it would be `tools/scripts`.

Is there a better location ?

AmirTallap commented:
> It seems the commit totals are not 100% correct.
> 
> @AmirTallap would you like to review this PR ?
> 
> Is there an easier way to do this ?

Checking!

BDadmehr0 committed:
Delete generate_authors.py

BDadmehr0 committed:
Create generate_authors.py to tools/scripts

BDadmehr0 committed:
Update generate_authors.py

Fix recoverable subprocess errors and silence unrecoverable ones

BDadmehr0 commented:
I fixed all the Bandit errors that were resolvable, but Bandit raises default security warnings for subprocess (B603 and B404), which don't have alternative solutions, so we need to silence them. @jbampton 

https://github.com/apache/sedona/pull/1895/commits/b3d9a634ee8f4e1164dff25eaa4f51ba4896773d

jbampton commented:
Hey @BDadmehr0 thanks for all of this.

Pretty sure the commit totals are not correct for some contributors

BDadmehr0 committed:
Merge branch 'apache:master' into AUTHORS

BDadmehr0 commented:
> Hey @BDadmehr0 thanks for all of this.
> 
> Pretty sure the commit totals are not correct for some contributors

Hello, sorry for the delay.

I've improved my script, but I need to make sure the current script isn't working correctly. If you're referring to the file https://github.com/apache/sedona/pull/1895/files#diff-ab6af77435f58cc0c9d4c31dfe05656e45187cc7c7fc02aada401a7642125463, I should mention that this script needs to be run in a cloned environment of https://github.com/apache/sedona/ on the main branch that has been recently updated to correctly show the commits.

Please test this and let me know.

BDadmehr0 commented:
Why do I want this? Because my improved script also doesn't fetch the latest commits in my cloned branch. @jbampton

BDadmehr0 commented:
@jbampton Issues fixed. Pre-commit passes and duplicates removed. Please review again when you can. Thanks!
